platform/security/core/security-core-services/src/main/resources/OSGI-INF/metatype/metatype.xml

```diff
@@ -38,11 +38,22 @@
           default="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"/>
     </OCD>
 
+    <OCD name="Session" id="ddf.security.http.impl.HttpSessionFactory">
+        <AD description="Specifies the length of inactivity (in minutes) between client requests before the servlet container will invalidate the session (this applies to all client sessions).
+            This value must be 2 minutes or greater, as users are warned when only 1 minute remains. If a value of less than 2 minutes is used, the timeout is set to the default time of 31 minutes."
```

Comment:
> This value must be 2 minutes or greater, as users are warned when only 1 minute remains. If a value of less than 2 minutes is used, the timeout is set to the default time of 31 minutes.

✏️ I don't think either of these statements apply to this config

Reply:
I think they do.

Reply:
Maybe I'm missing something, then. Can you please clarify?

> This value must be 2 minutes or greater, as users are warned when only 1 minute remains.
That's the session inactivity timeout notice, which isn't affected by this config. https://github.com/codice/ddf/blob/master/platform/platform-configuration/src/main/resources/OSGI-INF/metatype/metatype.xml

> If a value of less than 2 minutes is used, the timeout is set to the default time of 31 minutes.
I don't see where that check is performed in this code, here. Looks like it just accepts the value you set, even if it's less than 2 minutes. Again, I think this statement only applies to the platform session timeout config

Reply:
It is affected by both. If the total session length is also less than that, it will cause the same issues.

Reply:
It shouldn't be affected by both. This session auto renew utility keeps the session alive regardless of duration until the user has been inactive for the specified time: https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/singletons/session-auto-renew.js. If this timeout causes the described issues then auto-renewal is broken.

And again, I don't see where the 2 minute check is performed. This is where it's performed for the Platform UI config: https://github.com/codice/ddf/blob/master/platform/platform-configuration/src/main/java/org/codice/ddf/configuration/PlatformUiConfiguration.java#L188.
Where is that done here?